src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,476 @@
+            } else {
+                LOGGER.atDebug()
+                        .kv(LOG_KEY_REASON_CODES, subAckPacket.getReasonCodes())
+                        .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
+                        .kv(LOG_KEY_TOPIC, topic)
+                        .log("Failed to subscribe to topic");
+            }
+        } catch (ExecutionException e) {
+            LOGGER.atDebug().setCause(Utils.getUltimateCause(e)).kv(LOG_KEY_TOPIC, topic)
+                    .log("failed to subscribe");
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private boolean subscriptionIsSuccessful(SubAckPacket.SubAckReasonCode rc) {
+        return rc == SubAckPacket.SubAckReasonCode.GRANTED_QOS_0
+                || rc == SubAckPacket.SubAckReasonCode.GRANTED_QOS_1
+                || rc == SubAckPacket.SubAckReasonCode.GRANTED_QOS_2;
+    }
+
+    private void unsubscribe(String topic) {
+        if (!client.getIsConnected()) {
+            return;
+        }
+        UnsubscribePacket unsubscribePacket =
+                new UnsubscribePacket.UnsubscribePacketBuilder().withSubscription(topic).build();
+        LOGGER.atDebug().kv(LOG_KEY_TOPIC, topic).log("Unsubscribing from MQTT topic");
+        try {
+            UnsubAckPacket unsubAckPacket = client.unsubscribe(unsubscribePacket).get();
+            if (!unsubAckPacket.getReasonCodes().contains(UnsubAckPacket.UnsubAckReasonCode.SUCCESS)) {
+                LOGGER.atDebug()
+                        .kv(LOG_KEY_TOPIC, topic)
+                        .kv(LOG_KEY_REASON_STRING, unsubAckPacket.getReasonString())
+                        .kv(LOG_KEY_REASON_CODES, unsubAckPacket.getReasonCodes())
+                        .log("failed to unsubscribe");
+                return;
+            }
+            LOGGER.atDebug().kv(LOG_KEY_TOPIC, topic).log("Unsubscribed from topic");
+            synchronized (subscriptionsLock) {
+                subscribedLocalMqttTopics.remove(topic);
+            }
+        } catch (ExecutionException e) {
+            LOGGER.atDebug().setCause(Utils.getUltimateCause(e)).kv(LOG_KEY_TOPIC, topic)
+                    .log("failed to unsubscribe");
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private void unsubscribeAll() {
+        synchronized (subscriptionsLock) {
+            Set<String> toUnsubscribe = new HashSet<>(subscribedLocalMqttTopics);
+            LOGGER.atDebug().kv("topicsToUnsubscribe", toUnsubscribe).log("Unsubscribe from local MQTT topics");
+            toUnsubscribe.forEach(this::unsubscribe);
+            subscribedLocalMqttTopics.clear();
+        }
+    }
+
+    @Override
+    public void start()  {
+        client.start();
+    }
+
+    @Override
+    public void stop() {
+        synchronized (subscriptionsLock) {
+            cancelUpdateSubscriptionsTask();
+            unsubscribeAll();
```

Comment:
Do we actually need to unsubscribe in shutdown? I don't see what good it would do really.